inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -0,0 +1,308 @@
+        });
+
+        const that = this;
+
+        this.stompClient.onConnect = function (frame) {
+            that.connected = true;
+
+            //Receive username from inital message exchange header
+            const header = frame.headers;
+            let data: keyof typeof header;
+            for (data in header) {
+                that.client = header[data];
+                break;
+            }
+
+            //Receive project and document from URL
+
+            that.projectID = document.location.href.split("/")[5];
+            that.documentID = document.location.href.split("=")[1].split("&")[0];
+
+            // ------ DEFINE STANDARD SUBSCRIPTION CHANNELS WITH ACTIONS ------ //
+
+            that.stompClient.subscribe("/queue/new_document_for_client/" + that.client, function (msg) {
+                that.receiveNewDocumentMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            }, {id: "new_document"});
+
+            that.stompClient.subscribe("/queue/new_viewport_for_client/" + that.client, function (msg) {
+                that.receiveNewViewportMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            }, {id: "new_viewport"});
+
+            that.stompClient.subscribe("/queue/selected_annotation_for_client/" + that.client, function (msg) {
+                that.receiveSelectedAnnotationMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            }, {id: "selected_annotation"});
+
+            that.stompClient.subscribe("/queue/error_for_client/" + that.client, function (msg) {
+                that.receiveErrorMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            }, {id: "error_message"});
+        };
+
+
+        // ------ STOMP ERROR HANDLING ------ //
+
+        this.stompClient.onStompError = function (frame) {
+            console.log('Broker reported error: ' + frame.headers['message']);
+            console.log('Additional details: ' + frame.body);
+        };
+
+        this.stompClient.activate();
+    }
+
+    unsubscribe(aChannel: string) {
+        this.stompClient.unsubscribe(aChannel);
+    }
+
+    // ------ DISCONNECT -------- //
+    disconnect() {
+        if (this.connected) {
+            console.log("Disconnecting now");
+            this.connected = false;
+            this.stompClient.deactivate();
+        }
+    }
+
+    editAnnotation(aId, aAnnotationType) {
+        let json = {
+            username: this.client,
+            clientName: this.client,
+            project: this.projectID,
+            document: this.documentID,
+            annotationAddress: aId,
+            annotationType: aAnnotationType
+        };
+
+        //TODO correct edit
+        //this.stompClient.publish({destination: "/app/select_annotation_by_client", body: JSON.stringify(json)});
+    }
+
+    sendDocumentMessageToServer(aUsername, aDocument, aOffset, aOffsetType) {
```

Comment:
What does it mean to send a document to the server? Sounds to me like importing a document, but that is probably not what you mean, is it?